<?xml version="1.0"?>

<launch>

  <param name="vehicle/scripts_dir" type="str" value="$(find genesis_path_follower)/scripts/" />
  <!-- The trajectory can be generated using recorded time or a target velocity. -->
  <!-- Set track_using_time to True to allow for time-varying velocity. -->
  <!-- Set track_using_time to False to use target_vel (m/s) instead. -->
  <param name="vehicle/track_using_time" type="bool" value="True" />
  <param name="vehicle/target_vel" type="double" value="1.0" />

  <!-- Waypoints CSV to specify path. -->
  <!-- Set is_heading_info to true if Heading (N = 0) is used instead of Yaw (E = 0). -->
  <param name="vehicle/mat_waypoints" type="str" value="$(find genesis_path_follower)/paths/rightTurnRFSnitin.mat" />
  <param name="vehicle/is_heading_info" type="bool" value="False"/>
  <param name="vehicle/time_check_on" type="bool" value="False"/>

  <!-- RFS Coordinate System Definition -->
  <param name="vehicle/lat0" type="double" value="37.917929"/>
  <param name="vehicle/lon0" type="double" value="-122.331798"/>
  <param name="vehicle/yaw0" type="double" value="0.0"/>

  <!--...................................................................-->

  <!-- MPC, State Publisher, and Plotting Nodes -->
  <node ns="vehicle" pkg="genesis_path_follower" type="lk_cmd_pub.py" name="mpc_node" output="screen"/>
  <node ns="vehicle" pkg="genesis_path_follower" type="state_publisher.py" name="state_pub_node" output="log"/>
  <node ns="vehicle" pkg="genesis_path_follower" type="gps_plotter.py" name="plot_node" output="log"/>

  <!-- Recording Node -->
  <arg name="can_topics" default="/can_1/can_rx /can_1/can_tx /can_2/can_rx /rosout /rosout_agg"/>
  <arg name="gps_topics" default="/gps/fix /gps/odom /gps/vel /imu/data/"/>
  <arg name="vehicle_topics" default="/vehicle/imu /vehicle/state_est /vehicle/steering /vehicle/wheel_speeds /vehicle/target_path /vehicle/mpc_path"/>
  <arg name="control_topics" default="/control/enable_spas /control/enable_accel /control/accel /control/steer_angle"/>
  <arg name="perception_topics" default="/mando_camera/lane_left /mando_camera/lane_right"/>

  <node pkg="rosbag" type="record" name="rosbag_path_recorder" respawn="true" output="screen" 
        args="-a -o $(find genesis_path_follower)/bags/path_follow" />

</launch>
